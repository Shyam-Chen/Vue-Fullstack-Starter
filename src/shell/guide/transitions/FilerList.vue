<template>
  <v-layout column class="my-1">
    <div class="title mb-2">Filer List</div>

    <div>
      <v-btn @click="type = 'all'">All</v-btn>
      <v-btn @click="type = 'odd'">Odd</v-btn>
      <v-btn @click="type = 'even'">Even</v-btn>
    </div>

    <transition-group tag="ul" name="fade-group" class="filer-list">
      <li v-for="item in filerList" :key="item" class="filer-item">{{ item }}</li>
    </transition-group>
  </v-layout>
</template>

<script>
// @flow

type Data = {
  type: string,
  list: number[],
};

export default {
  data(): Data {
    return {
      type: 'all',
      list: Array(21)
        .fill(1)
        .map((x, y) => x + y), // 1 ~ 21
    };
  },
  computed: {
    filerList(): number[] {
      return this.list.filter(value => {
        if (this.type === 'odd') return value % 2 === 1;
        if (this.type === 'even') return value % 2 === 0;

        return this.list;
      });
    },
  },
};
</script>

<style scoped>
.filer-list {
  width: 21rem;
  padding: 0;
}

.filer-item {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: 2.5rem;
  height: 2.5rem;
  margin: 0.5rem;
  color: #fff;
  background: #009688;
  filter: drop-shadow(0 0 0.25rem #00796b);
}

.fade-group-enter-active,
.fade-group-leave-active,
.fade-group-move {
  transition: opacity 0.5s, transform 0.5s;
}

.fade-group-leave-active {
  position: absolute;
}

.fade-group-enter {
  transform: translateY(-1.25rem);
  opacity: 0;
}

.fade-group-leave-to {
  transform: translateY(1.25rem);
  opacity: 0;
}
</style>
