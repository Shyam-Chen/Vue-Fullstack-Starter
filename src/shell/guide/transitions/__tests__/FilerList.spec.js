import { shallowMount } from '@vue/test-utils';

import FilerList from '../FilerList';

describe('FilerList', () => {
  let [wrapper] = [];

  beforeEach(() => {
    wrapper = shallowMount(FilerList);
  });

  it('should render an initial component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render component with odd option', () => {
    wrapper.setData({ type: 'odd' });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render component with even option', () => {
    wrapper.setData({ type: 'even' });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
