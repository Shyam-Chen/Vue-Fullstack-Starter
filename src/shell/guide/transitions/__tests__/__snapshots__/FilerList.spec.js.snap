// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilerList should render an initial component 1`] = `
<v-layout-stub tag="div" column="" class="my-1">
  <div class="title mb-2">Filer List</div>
  <div>
    <v-btn-stub tag="button" activeclass="" type="button">All</v-btn-stub>
    <v-btn-stub tag="button" activeclass="" type="button">Odd</v-btn-stub>
    <v-btn-stub tag="button" activeclass="" type="button">Even</v-btn-stub>
  </div> <span tag="ul" name="fade-group" class="filer-list"><li class="filer-item">1</li><li class="filer-item">2</li><li class="filer-item">3</li><li class="filer-item">4</li><li class="filer-item">5</li><li class="filer-item">6</li><li class="filer-item">7</li><li class="filer-item">8</li><li class="filer-item">9</li><li class="filer-item">10</li><li class="filer-item">11</li><li class="filer-item">12</li><li class="filer-item">13</li><li class="filer-item">14</li><li class="filer-item">15</li><li class="filer-item">16</li><li class="filer-item">17</li><li class="filer-item">18</li><li class="filer-item">19</li><li class="filer-item">20</li><li class="filer-item">21</li></span>
</v-layout-stub>
`;

exports[`FilerList should render component with even option 1`] = `
<v-layout-stub tag="div" column="" class="my-1">
  <div class="title mb-2">Filer List</div>
  <div>
    <v-btn-stub tag="button" activeclass="" type="button">All</v-btn-stub>
    <v-btn-stub tag="button" activeclass="" type="button">Odd</v-btn-stub>
    <v-btn-stub tag="button" activeclass="" type="button">Even</v-btn-stub>
  </div> <span tag="ul" name="fade-group" class="filer-list"><li class="filer-item">2</li><li class="filer-item">4</li><li class="filer-item">6</li><li class="filer-item">8</li><li class="filer-item">10</li><li class="filer-item">12</li><li class="filer-item">14</li><li class="filer-item">16</li><li class="filer-item">18</li><li class="filer-item">20</li></span>
</v-layout-stub>
`;

exports[`FilerList should render component with odd option 1`] = `
<v-layout-stub tag="div" column="" class="my-1">
  <div class="title mb-2">Filer List</div>
  <div>
    <v-btn-stub tag="button" activeclass="" type="button">All</v-btn-stub>
    <v-btn-stub tag="button" activeclass="" type="button">Odd</v-btn-stub>
    <v-btn-stub tag="button" activeclass="" type="button">Even</v-btn-stub>
  </div> <span tag="ul" name="fade-group" class="filer-list"><li class="filer-item">1</li><li class="filer-item">3</li><li class="filer-item">5</li><li class="filer-item">7</li><li class="filer-item">9</li><li class="filer-item">11</li><li class="filer-item">13</li><li class="filer-item">15</li><li class="filer-item">17</li><li class="filer-item">19</li><li class="filer-item">21</li></span>
</v-layout-stub>
`;
